### PR TITLE
Solid js support

### PR DESCRIPTION
**Pull request by RicardoValero95**

### CODE REVIEW

**Comment by hannoeru** [comment]
Could you add some test for this?

**Comment by RicardoValero95** [comment]
I'll try to figure it out

**Comment by RicardoValero95** [comment]
I basically copied the vue test but skipped the markdown test

**Comment by hannoeru** [comment]
I think you can just remove the markdown test.
But why you create a new example just for test? I prefer remove the exist Solid example and replace it with test example.
Also you are missing generation test in `generate.spec.ts`.

File: examples/solid-test/src/features/dashboard/pages/dashboard.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+/* TODO
+<route>
+{
+  meta: {
+    requiresAuth: true
+  }
+}
+</route>
```
Comment (hannoeru) [inline examples/solid-test/src/features/dashboard/pages/dashboard.tsx:15]: Parsing HTML block in JSX files is not supported. This will not work.

**Comment by RicardoValero95** [comment]
I don't get why it fails, any insight is appreciated

**Comment by hannoeru** [comment]
Can you add permission to let me edit this PR?

**Comment by RicardoValero95** [comment]
I believe you already had permission
image

if this is not it, then please guide me, I'm learning

File: src/resolvers/solid.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  rawRoute: string
+  path: string
+  children?: Route[]
+  component?: string
+  element?: string
+}
+
+export type PrepareRoute = Omit<Optional<Route, 'rawRoute' | 'path'>, 'children'> & {
+  children?: PrepareRoute[]
+}
```
Comment (hannoeru) [inline src/resolvers/solid.ts:21]: I think is better to rename this two types like this, also in other resolves.
```suggestion
export interface SolidRouteBase {
  rawRoute: string
  path: string
  children?: Route[]
  component?: string
  element?: string
}

export type SolidRoute = Omit<Optional<SolidRouteBase, 'rawRoute' | 'path'>, 'children'> & {
  children?: PrepareRoute[]
}
```

File: src/resolvers/vue.ts [hunk 3]
```diff
@@ -18,12 +18,12 @@ interface Route {
   rawRoute: string
 }
 
-type PrepareRoutes = Omit<Optional<Route, 'rawRoute' | 'name'>, 'children'> & {
+export type PrepareRoutes = Omit<Optional<Route, 'rawRoute' | 'name' | 'component'>, 'children'> & {
```
Comment (hannoeru) [inline src/resolvers/vue.ts:21]: Why you add `'component'` to options?

File: src/resolvers/vue.ts [hunk 4]
```diff
@@ -99,16 +99,16 @@ export async function resolveVueRoutes(ctx: PageContext) {
...
       if (parent) {
-        // Make sure children exits in parent
+        // Make sure children exist in parent
         parent.children = parent.children || []
         // Append to parent's children
         parentRoutes = parent.children
         // Reset path
         route.path = ''
-      } else if (normalizedName.toLowerCase() === 'index') {
+      } else if (normalizedPath === 'index') {
```
Comment (hannoeru) [inline src/resolvers/vue.ts:108]: https://github.com/hannoeru/vite-plugin-pages/pull/77

File: src/resolvers/vue.ts [hunk 5]
```diff
@@ -99,16 +99,16 @@ export async function resolveVueRoutes(ctx: PageContext) {
...
         if (!route.path)
           route.path = '/'
-      } else if (normalizedName.toLowerCase() !== 'index') {
+      } else if (normalizedPath !== 'index') {
```
Comment (hannoeru) [inline src/resolvers/vue.ts:111]: https://github.com/hannoeru/vite-plugin-pages/pull/77

File: src/types.ts [hunk 6]
```diff
@@ -58,15 +60,15 @@ interface Options {
...
   /**
    * Custom generated routes
    */
-  onRoutesGenerated?: (routes: any[]) => any[] | void | Promise<any[] | void>
+  onRoutesGenerated?: (routes: PrepareRoute[]) => Route[] | Promise<Route[] | void>
   /**
    * Custom generated client code
    */
-  onClientGenerated?: (clientCode: string) => string | void | Promise<string | void>
+  onClientGenerated?: (clientCode: string) => string | Promise<string | void>
```
Comment (hannoeru) [inline src/types.ts:71]: We don't have consistent field in route to let typescript know what type is this route, so currently we can't resolve these types, you can check generate test file, there is a lot of typescript errors, that is why I keep it `any`.
But maybe we can export these types for users.

Comment (RicardoValero95) [reply src/types.ts:71]: You are right

**Pull request merged**